crates/bevy_ui/src/render/mod.rs

```diff
@@ -522,110 +523,18 @@ pub fn extract_uinode_borders(
-                transform,
-                color: border_color.0.into(),
-                rect: Rect {
-                    max: uinode.size(),
-                    ..Default::default()
-                },
-                image,
-                atlas_scaling: None,
-                clip: clip.map(|clip| clip.clip),
-                flip_x: false,
-                flip_y: false,
-                camera_entity,
-                border_radius,
-                border,
-                node_type: NodeType::Border,
-            },
-        );
-    }
-}
-
-pub fn extract_uinode_outlines(
-    mut commands: Commands,
-    mut extracted_uinodes: ResMut<ExtractedUiNodes>,
-    default_ui_camera: Extract<DefaultUiCamera>,
-    uinode_query: Extract<
-        Query<(
-            &Node,
-            &GlobalTransform,
-            &ViewVisibility,
-            Option<&CalculatedClip>,
-            Option<&TargetCamera>,
-            &Outline,
-        )>,
-    >,
-) {
-    let image = AssetId::<Image>::default();
-    for (node, global_transform, view_visibility, maybe_clip, camera, outline) in &uinode_query {
-        let Some(camera_entity) = camera.map(TargetCamera::entity).or(default_ui_camera.get())
-        else {
-            continue;
-        };
-
-        // Skip invisible outlines
-        if !view_visibility.get()
-            || outline.color.is_fully_transparent()
-            || node.outline_width == 0.
-        {
-            continue;
-        }
-
-        // Calculate the outline rects.
-        let inner_rect = Rect::from_center_size(Vec2::ZERO, node.size() + 2. * node.outline_offset);
-        let outer_rect = inner_rect.inflate(node.outline_width());
-        let outline_edges = [
-            // Left edge
-            Rect::new(
-                outer_rect.min.x,
-                outer_rect.min.y,
-                inner_rect.min.x,
-                outer_rect.max.y,
-            ),
-            // Right edge
-            Rect::new(
-                inner_rect.max.x,
-                outer_rect.min.y,
-                outer_rect.max.x,
-                outer_rect.max.y,
-            ),
-            // Top edge
-            Rect::new(
-                inner_rect.min.x,
-                outer_rect.min.y,
-                inner_rect.max.x,
-                inner_rect.min.y,
-            ),
-            // Bottom edge
-            Rect::new(
-                inner_rect.min.x,
-                inner_rect.max.y,
-                inner_rect.max.x,
-                outer_rect.max.y,
-            ),
-        ];
 
-        let world_from_local = global_transform.compute_matrix();
-        for edge in outline_edges {
-            if edge.min.x < edge.max.x && edge.min.y < edge.max.y {
+        // don't extract border if no border or the node is zero-sized (a zero sized node can still have an outline).
+        if uinode.size().x > 0. && uinode.size().y > 0. && border != [0.; 4] {
```

Comment:
I'd still be happier if the condition was split into two nice booleans, maybe the following?
```rust
let has_border = border != [0.; 4];
let node_has_size = uinode.size().x > 0. && uinode.size().y > 0.;
```